### ISSUE TITLE
Selection restoring – take two

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-undo#64]
After https://github.com/ckeditor/ckeditor5-engine/pull/977 and https://github.com/ckeditor/ckeditor5-undo/pull/63 we basically need to rewrite the selection restoring mechanism.

We need to think again when the selection to be restored needs to be remembered, how it can be transformed (ideally, using `Range#getTransformedByDeltas()`), by which deltas, and how undo and redo commands need to communicate (and if they need to communicate at all).

Basically, the whole code can be removed and written from the beginning.

**Comment by Reinmar** [comment]
This needs to wait for all the refactoings to finish.
